SecuritySystemUWP/SecuritySystemUWP/Camera/Webcam.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Windows.Devices.Gpio;
+using Windows.Storage;
+using System.Collections.Generic;
+using Microsoft.Maker.Devices.Media.UsbCamera;
+using Microsoft.Maker.Devices.Gpio.PirSensor;
+
+namespace SecuritySystemUWP
+{
+    public class Webcam : ICamera
+    {
+        private UsbCamera webcam;
+        private PirSensor pirSensor;
+        private bool isCapturing;
```

Comment:
Spoke to Apu, this was used to prevent parallel invocations of the TakePhotoAsync method (since this causes exceptions in the CapturePhoto API . Suggestion is to use Monitor.TryEnter/Exit inside this method to check that the codepath is currently not being executed.